### PR TITLE
RavenDB-17687 - debugability layer to allow complex queries inspection

### PR DESCRIPTION
**Pull request by redknightlois**
Pushing for retrieving feedback only. cc @ayende @arekpalinski @ppekrol @maciejaszyk

### CODE REVIEW

File: src/Corax/Queries/BinaryMatch.Boolean.cs [hunk 1]
```diff
@@ -42,6 +42,17 @@ static int AndWith(ref BinaryMatch<TInner, TOuter> match, Span<long> matches)
                 return outer.AndWith(matches.Slice(0, results));
             }
 
+            static QueryInspectionNode InspectFunc(ref BinaryMatch<TInner, TOuter> match)
+            {
+                return new QueryInspectionNode($"{nameof(BinaryMatch)} [And]",
```
Comment (ayende) [inline src/Corax/Queries/BinaryMatch.Boolean.cs:47]: You didn't implement this for OR?

**Review by ayende — COMMENTED** [review]
Looks good, but also need to turn that into a string, so we can very quickly inspect this

File: src/Corax/Queries/BinaryMatch.Boolean.cs (hunk 1 above)
Comment (redknightlois) [reply src/Corax/Queries/BinaryMatch.Boolean.cs:47]: Yes.
https://github.com/ravendb/ravendb/pull/13270/files/51fa09050f5e2500b16b71ade428465cd19ff2a3#diff-387c28ce8317892df9dd1c7dda739e8826bb775064d6871f3e5e68ebd446f7fcR171

**Comment by redknightlois** [comment]
Yes, that's the next step. Wanted to know first if there is any use case that I would be missing.

**Comment by arekpalinski** [comment]
LGTM at this point

**Review by arekpalinski — APPROVED** [review]

**Comment by arekpalinski** [comment]
@redknightlois can you please solve the conflicts?

**Comment by arekpalinski** [comment]
Please make sure that you're rebased against `feature/corax` branch

**Pull request merged**